### PR TITLE
feat(clickhouse sink): make `database` and `table` templateable

### PR DESCRIPTION
**Pull request by dsmith3197**
Closes: https://github.com/vectordotdev/vector/issues/16329

### CODE REVIEW

File: src/sinks/clickhouse/config.rs [hunk 1]
```diff
@@ -90,25 +90,30 @@ impl SinkConfig for ClickhouseConfig {
...
         let service = ServiceBuilder::new()
             .settings(request_limits, ClickhouseRetryLogic::default())
             .service(service);
 
         let batch_settings = self.batch.into_batcher_settings()?;
+        let database = self.database.clone().unwrap_or_else(|| {
+            "default"
+                .try_into()
+                .expect("'default' should be a valid template")
+        });
```
Comment (spencergilbert) [inline src/sinks/clickhouse/config.rs:108]: I think I'm missing why we only do this for the `database` here, but not the `table`?

Comment (dsmith3197) [reply src/sinks/clickhouse/config.rs:108]: `database` is optional whereas `table` is required in the config :)

Comment (spencergilbert) [reply src/sinks/clickhouse/config.rs:108]: Ugh, I knew it'd be obvious 🤦

**Review by spencergilbert — APPROVED** [review]

File: vdev/src/testing/integration.rs [hunk 2]
```diff
@@ -179,7 +179,10 @@ impl Compose {
                 // Inject the networks block
                 config.networks.insert(
                     "default".to_string(),
-                    BTreeMap::from_iter([("name".to_string(), network.clone())]),
+                    BTreeMap::from_iter([
+                        ("name".to_string(), network.clone()),
+                        ("external".to_string(), "true".to_string()),
```
Comment (dsmith3197) [inline vdev/src/testing/integration.rs:184]: `external: true` was required to get the integration tests running after I upgraded Docker locally. I'll verify this doesn't break anything in CI.

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**